boards/arm/mps2_an521/doc/index.rst

```diff
@@ -51,21 +51,64 @@ The BOARD options are summarized below:
 | mps2_an521_remote    | For building firmware on CPU1                         |
 +----------------------+-------------------------------------------------------+
 
+Memory Partitioning
+===================
+
+The AN521 has 4MB allocated for code space, and 4MB for SRAM. These memory
+regions are shared across both cores, and are aliased in both secure and
+non-secure regions, where the secure memory alias has an offset of
+0x10000000 relative to non-secure.
+
+The following memory map and partitioning schemes are used by default, where
+the offset value is the offset from the base of the 4MB code or SRAM block,
+ignoring the S/NS alias difference.
+
++-------------------+-----+----------------+----------------+------------+
+| Board             | CPU | Code (Offset)  | SRAM (Offset)  | S/NS Alias |
++===================+=====+================+================+============+
+| mps2_an521        | 0   | 4MB (0)        | 4MB (0)        | S          |
++-------------------+-----+----------------+----------------+------------+
+| mps2_an521_ns     | 0   | 512KB (512KB)  | 512KB (512KB)  | NS         |
++-------------------+-----+----------------+----------------+------------+
+| mps2_an521_remote | 1   | 468KB (3628KB) | 512KB (1536KB) | NS         |
++-------------------+-----+----------------+----------------+------------+
+
+The ``mps2_an521_ns`` board target is intended to be used with TF-M, with the
+Zephyr memory map matching the AN521 memory map defined upstream in TF-M. TF-M
+boots the secure processing environment before initialising Zephyr in the
+non-secure processing environment. The non-secure Zephyr image is offset to
+make room for the secure bootloader, and the secure firmware (TF-M), resulting
+in a starting address of 0x00100000. SRAM begins with a 512KB offset at
+0x28100000.
+
+The ``mps2_an521_remote`` board target is setup for the second core on the
+AN521, using the final 468KB code memory in the 4MB code block. This value
+is chosen to maintain compatibility with TF-M, which marks that final 468KB
```

Comment:
I've already sent a patch to TF-M for fix their memory map: https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/17645

Reply:
I took the `Unused` start address from the worst-case scenario, which is in the second memory map, depending on the way compilation is configured for TF-M:

```
 * Flash layout on MPS2 AN521 with BL2 (single image boot):
 *
 * 0x0000_0000 BL2 - MCUBoot (0.5 MB)
 * 0x0008_0000 Primary image area (1 MB):
 *    0x0008_0000 Secure     image primary
 *    0x0010_0000 Non-secure image primary
 * 0x0018_0000 Secondary image area (1 MB):
 *    0x0018_0000 Secure     image secondary
 *    0x0020_0000 Non-secure image secondary
 * 0x0028_0000 Scratch area (1 MB)
 * 0x0038_0000 Protected Storage Area (20 KB)
 * 0x0038_5000 Internal Trusted Storage Area (16 KB)
 * 0x0038_9000 OTP / NV counters area (8 KB)
 * 0x0038_B000 Unused (468 KB)
```

In this layout, our NS image still starts at 0x0010_0000, but we have less unused space since we need more flash memory reserved if we have to roll back an entire (1MB) image at once, not just a 512 KB image.

Reply:
@mbolivar-nordic I've fixed the other issues (thanks for the thorough review), please let me know if you disagree with my approach to assume a worst-case scenario for flash memory layout in TF-M depending on the firmware update strategy you select.